src/system_database.ts

```diff
@@ -553,6 +554,69 @@ export class PostgresSystemDatabase implements SystemDatabase {
     }
   }
 
+  async forkWorkflow(originalWorkflowID: string, forkedWorkflowId: string, startStep: number = 1): Promise<string> {
```

Comment:
Let's take this into a separate PR because this PR is only for resume/restart (fork from beginning). Also in TS, step starts from 0.

```suggestion
  async forkWorkflow(originalWorkflowID: string, forkedWorkflowId: string): Promise<string> {
```